crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -867,129 +869,139 @@ where
         //     })
         //     .collect();
 
-        // Index epoch
-        let epoch_index = if checkpoint.epoch == 0 && checkpoint.sequence_number == 0 {
-            // very first epoch
-            let system_state: SuiSystemStateSummary = self
-                .http_client
-                .get_latest_sui_system_state()
-                .await
-                .map_err(|e| {
-                    IndexerError::FullNodeReadingError(format!(
-                        "Failed to get latest system state with error {:?}",
-                        e
-                    ))
-                })?;
-            let validators = system_state
-                .active_validators
-                .iter()
-                .map(|v| (system_state.epoch, v.clone()).into())
-                .collect();
-
-            Some(TemporaryEpochStore {
-                last_epoch: None,
-                new_epoch: DBEpochInfo {
-                    epoch: 0,
-                    first_checkpoint_id: 0,
-                    epoch_start_timestamp: system_state.epoch_start_timestamp_ms as i64,
-                    ..Default::default()
-                },
-                system_state: system_state.into(),
-                validators,
-            })
-        } else if let Some(end_of_epoch_data) = &checkpoint.end_of_epoch_data {
-            // Find system state object
-            let system_state: SuiSystemStateSummary = self
-                .http_client
-                .get_latest_sui_system_state()
-                .await
-                .map_err(|e| {
-                    IndexerError::FullNodeReadingError(format!(
-                        "Failed to get latest system state with error {:?}",
-                        e
-                    ))
-                })?;
-
-            let epoch_event = transactions.iter().find_map(|tx| {
-                tx.events.data.iter().find(|ev| {
-                    ev.type_.address == SUI_SYSTEM_ADDRESS
-                        && ev.type_.module.as_ident_str() == ident_str!("sui_system_state_inner")
-                        && ev.type_.name.as_ident_str() == ident_str!("SystemEpochInfoEvent")
+        // NOTE: Index epoch when object index has reached the same checkpoint,
```

Comment:
This comment was a bit confusing since to my understanding `index_checkpoint` just cares about whether to index epoch depending on the truthiness of `index_epoch`. Maybe this can be on L248 instead? I'm probably just missing context though

Reply:
we split the checkpoint indexing into tx checkpoint indexing & obj checkpoint indexing, epoch indexing depends on the obj checkpoint index, as they need the latest system state, which is stored in the obj. I will change the wording to object checkpoint index then.